[docs] Make clear that the client is the one logging artifacts to the artifact store (#3046)
mlflow/mlflow: docs/source/tracking.rst

running ``mlflow run`` locally), but when running a server, make sure that this points to a
persistent (that is, non-ephemeral) file system location.

> Artifact Stores

Reviewer: Thanks for fixing this duplicated heading! Could you move the new, consolidated `Artifact Stores` section up here? It should go above `File store performance`, which is not as useful to know about, and `Deletion behavior`, which mentiones the artifact store.

Author: updated

persistent (that is, non-ephemeral) file system location.

Artifact Stores
~~~~~~~~~~~~~~~~

.. contents:: In this section:
  :local:
  :depth: 1

> <<<<<<< HEAD

Reviewer: Please resolve this git conflict

In addition to local file paths, MLflow supports the following storage systems as artifact
stores: Amazon S3, Azure Blob Storage, Google Cloud Storage, SFTP server, and NFS.

> The artifact store is a location suitable for large data (such as an S3 bucket or shared NFS

Reviewer: Could you move this block above the `In addition to local file paths, MLflow supports the following storage systems as artifact stores: Amazon S3, Azure Blob Storage, Google Cloud Storage, SFTP server, and NFS.`? We should describe what the artifact store is before mentioning the supported storage systems.

Author: updated

        mlflow_set_experiment("/my-experiment")
        mlflow_log_param("a", "1")

Note that to use the artifact storage capabilities of a remote server, the client need a direct access to the artifact store.
The client is the one pushing directly artifacts to the artifact storage, and not through the tracking server.

> How to setup credentials for the artifact storage depends on the :ref:`Artifact Stores <artifact-stores>` technology setup for the server.

Reviewer 2: ```suggestion
How to setup credentials for the artifact storage depends on the :ref:`Artifact Stores <artifact-stores>` technology setup for the server. Same holds for the definition of write options, like giving other users (e.g. the Tracking Server UI) permissions to read the artifacts.  
```
I want to hint at the possible problem that the tracking server can't read the files written by the client, for example when the client pushed files to S3 without [EXTRA_ARGS](https://www.mlflow.org/docs/latest/tracking.html#amazon-s3)

        mlflow_set_experiment("/my-experiment")
        mlflow_log_param("a", "1")

> Note that to use the artifact storage capabilities of a remote server, the client need a direct access to the artifact store. The client is the one pushing directly artifacts to the artifact storage, and not through the tracking server. How to setup credentials for the artifact storage depends on the :ref:`Artifact Stores <artifact-stores>` technology setup for the server. Note that some technology choices impose that credential will transit in clear text on the network.

Reviewer: Any reason to put this here, rather than in the Artifact Store section?

Reviewer: (also I think the test failure may be because of an invalid ref)

Author: the reason was your own suggestion in the original issue ([here](https://github.com/mlflow/mlflow/issues/2834#issuecomment-639892606)) :wink: 

yes it may be that my removal of the duplicated sections break things... is there a way to go for solving this ?

Reviewer: Ha, my mistake. This is a reasonable place to put this note. 

For debugging the failure, I think you can run Sphinx locally with this command: https://github.com/mlflow/mlflow/blob/master/.circleci/config.yml#L33

        mlflow_set_experiment("/my-experiment")
        mlflow_log_param("a", "1")

> Note that to use the artifact storage capabilities of a remote server, the client need a direct access to the artifact store.
> The client is the one pushing directly artifacts to the artifact storage, and not through the tracking server.

Reviewer: ```suggestion
To use the artifact storage capabilities of a remote server, the client needs direct access to the artifact store.
The client directly pushes artifacts to the artifact storage; it does not proxy them through the tracking server.
```

How to setup credentials for the artifact storage depends on the :ref:`Artifact Stores <artifact-stores>` technology setup for the server.

> Same holds for the definition of write options, like giving other users (e.g. the Tracking Server UI) permissions to read the artifacts.  

Reviewer: Do you think we need the above two lines?

Author: this was a suggestion by @fabboe earlier in this thread. The idea was to hint at the possible problem that the tracking server can't read the files written by the client, for example when the client pushed files to S3 without EXTRA_ARGS.

> Note that some technology choices impose that credential will transit in clear text on the network.

Reviewer: Could you elaborate this line? (include in what situation will the credentials be sent over the network in clear text)

Author: This is true at least for the FTP server, I admit I did not think all options thoroughly, so I thought this would be enough.

About the comment: did you want me to elaborate here or within the PR itself ?

Reviewer: Within the PR please. I think the docs readers will have the same question I do. (and if you don't know the answer / can't find it easily, feel free to remove this line)
